exec/java-exec/src/main/java/org/apache/drill/exec/coord/ClusterCoordinator.java

```diff
@@ -60,16 +62,23 @@
   public abstract DistributedSemaphore getSemaphore(String name, int maximumLeases);
 
   /**
+   * Returns a new {@link TransientStore store} instance with the given {@link TransientStoreConfig configuration}.
+   * @param config  store configuration
+   * @param <V>  value type for this store
+   */
+  public abstract <V> TransientStore<V> newTransientStore(TransientStoreConfig<V> config);
+
+  /**
    * Actions to take when there are a set of new de-active drillbits.
    * @param unregisteredBits
    */
-  public void drillbitUnregistered(Set<DrillbitEndpoint> unregisteredBits) {
+  protected void drillbitUnregistered(Set<DrillbitEndpoint> unregisteredBits) {
```

Comment:
Wrong use of this method outside of subclasses could easily mess the system up. To my understanding, this method should only be visible to implementor who is the ultimate authority to fire listeners on topology changes, so protected seems to make more sense here.